cancer_ai/rewarder_test.py

```diff
@@ -0,0 +1,301 @@
+        expected_share_leader_1_competition_2 = base_share
+        expected_share_leader_2 = base_share
+
+        remaining_share_leader_1_competition_1 = base_share - expected_share_leader_1_competition_1
+        # The competitors of competition 2 and 3 (including leader-1) get the distributed reduction    
+        expected_score_leader_1 = expected_share_leader_1_competition_1 + expected_share_leader_1_competition_2\
+              + remaining_share_leader_1_competition_1 / 2
+        expected_score_leader_2 = expected_share_leader_2 + remaining_share_leader_1_competition_1 / 2
+
+        self.assertAlmostEqual(rewarder.scores["leader-1"], expected_score_leader_1, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-2"], expected_score_leader_2, places=2)
+
+    def test_five_competitions_three_competitors_two_repeating(self):
+        """Test case 9: 5 competitions with 3 competitors, 2 of them are repeating"""
+        competitions_leaders = {
+            "competition-1": CompetitionLeader(hotkey="leader-1", leader_since=datetime.now() - timedelta(days=21)),  # 10% reduction
+            "competition-2": CompetitionLeader(hotkey="leader-2", leader_since=datetime.now() - timedelta(days=10)),  # No reduction
+            "competition-3": CompetitionLeader(hotkey="leader-1", leader_since=datetime.now()),  # No reduction
+            "competition-4": CompetitionLeader(hotkey="leader-3", leader_since=datetime.now() - timedelta(days=35)),  # 30% reduction
+            "competition-5": CompetitionLeader(hotkey="leader-2", leader_since=datetime.now())  # No reduction
+        }
+        scores = {"leader-1": 0, "leader-2": 0, "leader-3": 0}
+        rewarder_config = RewarderConfig(competition_leader_mapping=competitions_leaders, scores=scores)
+        rewarder = Rewarder(config=rewarder_config)
+
+        rewarder.update_scores()
+
+        base_share = 1 / 5
+        reduction_factor_leader_1_competition_1 = 0.9  # 10% reduction for 21 days
+        reduction_factor_leader_3_competition_4 = 0.7  # 30% reduction for 35 days
+
+        # Calculate expected shares for each leader
+        expected_share_leader_1_competition_1 = base_share * reduction_factor_leader_1_competition_1
+        expected_share_leader_1_competition_3 = base_share
+        expected_share_leader_2_competition_2 = base_share
+        expected_share_leader_2_competition_5 = base_share
+        expected_share_leader_3_competition_4 = base_share * reduction_factor_leader_3_competition_4
+
+        remaining_share_leader_1_competition_1 = base_share - expected_share_leader_1_competition_1
+        remaining_share_leader_3_competition_4 = base_share - expected_share_leader_3_competition_4
+
+        # Calculate final scores with distributed reduction shares
+        expected_score_leader_1 = expected_share_leader_1_competition_1 + expected_share_leader_1_competition_3\
```

Comment:
you shouldn't calculate, if you do, clear on top, but based on simple values, the less programmatically the better